go/beacon/init.go

```diff
@@ -6,43 +6,19 @@ import (
 	"fmt"
 	"strings"
 
-	flag "github.com/spf13/pflag"
-	"github.com/spf13/viper"
-
 	"github.com/oasislabs/oasis-core/go/beacon/api"
 	"github.com/oasislabs/oasis-core/go/beacon/tendermint"
 	epochtime "github.com/oasislabs/oasis-core/go/epochtime/api"
-	commonFlags "github.com/oasislabs/oasis-core/go/oasis-node/cmd/common/flags"
 	"github.com/oasislabs/oasis-core/go/tendermint/service"
 )
 
-const (
-	// CfgDebugDeterministic is the enable deterministic beacon output flag.
-	CfgDebugDeterministic = "beacon.debug.deterministic"
-)
-
-// Flags has the configuration flags.
-var Flags = flag.NewFlagSet("", flag.ContinueOnError)
-
 // New constructs a new Backend based on the configuration flags.
 func New(ctx context.Context, timeSource epochtime.Backend, tmService service.TendermintService) (api.Backend, error) {
-	backend := commonFlags.ConsensusBackend()
+	backend := tmService.GetGenesis().ConsensusBackend
```

Comment:
This looks funny as you are querying the tendermint service to give you the name of the consensus backend :grin: But this should all be fixed once we do #1879 as then the consensus backend will create these. Maybe just add an `XXX` comment that this will be improved in oasis-core#1879.